src/encoder.rs

```diff
@@ -3449,9 +3419,37 @@ pub fn encode_show_existing_frame<T: Pixel>(
   fi: &FrameInvariants<T>, fs: &mut FrameState<T>, inter_cfg: &InterConfig,
 ) -> Vec<u8> {
   debug_assert!(fi.show_existing_frame);
+  let obu_extension = 0;
+
   let mut packet = Vec::new();
 
-  write_obus(&mut packet, fi, fs, inter_cfg).unwrap();
+  if fi.frame_type == FrameType::KEY {
+    write_key_frame_obus(&mut packet, fi, obu_extension).unwrap();
+  }
+  let mut buf1 = Vec::new();
```

Comment:
Not really in scope for this PR since it matches the previous state of the code, but I'd like some clarity about why we have two buffers and writers, through renaming the variables or comments. It's not clear at a glance why this code is written the way it is.